src/blueprint/README.md

```diff
@@ -0,0 +1,153 @@
+# Azure CLI blueprint Extension #
+This package is for the 'blueprint' extension, i.e. 'az blueprint'. More info on what is [Blueprint](https://docs.microsoft.com/azure/governance/blueprints/overview).
+
+### How to use ###
+Install this extension using the below CLI command
+```
+az extension add --name blueprint
+```
+
+Blueprint can be scoped in a subscription or management group, which is a group of subscriptions. You can use management group by specify `--management-group`, or use subscription by specify `--subscription.` If both parameters are omitted, the command will use your default subscription. You can set it by running:
+```
+az account set --subscription subscription_id
+```
+
+### Included Features
+#### Blueprint Definition Management:
+*Examples:*
+
+##### Import Blueprint Definition and Artifacts Settings
+
+```
+az blueprint import \
+--name blueprintName \
+--input-path "/path/to/blueprint/directory"
+
+```
+
+In the input directory, there should be a file named "blueprint.json" that has blueprint definition and parameters for artifacts. There should be a subdirectory named "artifacts" and it should contain files for artifact definition. Examples can be found [here]().
+
+The import command will overwrite defnitions for blueprint and artifacts if a blueprint with the same name already exists.
+
+##### Create a Blueprint with Parameters
+
+```
+az blueprint create \
+    --name blueprintName \
+    --description "An example blueprint." \
+    --target-scope "subscription" \
+    --parameters @/path/to/blueprint_params.json
+```
+An example blueprint_params.json may look like this:
+```json
+{
+    "MyRoleName_RoleAssignmentName": {
+        "type": "array",
+        "displayName": "[User group or application name] ([User group or application name] : Reader)",
+        "strongType": "PrincipalId"
+    },
+    "MyPolicyName_Members": {
+      "type": "string",
+      "displayName": "Members",
+      "allowedValues": []
+    }
+}
+```
+The paramters are for artifacts which will be added in below commands.
+
+##### Add a Resource Group in the Blueprint
+```
+az blueprint resource-group create \
+    --blueprint-name blueprintName \
+    --artifact-name myRgArt
+```
+
+##### Add a Role Assignment Artifact
+```
+az blueprint artifact role create \
+    --blueprint-name blueprintName \
```

Comment:
From CLI general guideline, for the referencing resource, `--blueprint` is enough to support both name and id, right?

Reply:
blueprint resource id can be under management group, currently parse_resource_id does not support it. I've submitted an [issue](https://github.com/Azure/msrestazure-for-python/issues/139). So now it's name only.

Reply:
May I know how blueprint resource id looks like ? and is it different for management scope and subscription scope?

Reply:
They are different:
`/providers/Microsoft.Management/managementGroups/CliMgmtGroup/providers/Microsoft.Blueprint/blueprints/group-bp`
or
`/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/providers/Microsoft.Blueprint/blueprints/first-blueprint`